packages/expo-router/src/testing-library/index.tsx

```diff
@@ -47,30 +48,20 @@ export { MockContextConfig, getMockConfig, getMockContext };
 
 export function renderRouter(
   context: MockContextConfig = './app',
-  { initialUrl = '/', ...options }: RenderRouterOptions = {}
+  { initialUrl = '/', linking, ...options }: RenderRouterOptions = {}
 ): Result {
   jest.useFakeTimers();
 
   const mockContext = getMockContext(context);
 
-  // Reset the initial URL
-  setInitialUrl(initialUrl);
-
   // Force the render to be synchronous
   process.env.EXPO_ROUTER_IMPORT_MODE = 'sync';
   stateCache.clear();
 
-  let location: URL | undefined;
-
-  if (typeof initialUrl === 'string') {
-    location = new URL(initialUrl, 'test://');
-  } else if (initialUrl instanceof URL) {
-    location = initialUrl;
-  }
-
-  const result = render(<ExpoRoot context={mockContext} location={location} />, {
-    ...options,
-  });
```

Comment:
This logic has been moved inside `<ExpoRoot />` https://github.com/expo/expo/pull/27757/files#diff-999071a0dfb507af3e9096ca92f42027e98aa2a51d970a4fa955d069ea50447dR75-R97

`<ExpoRoot />` can now accept either a string or a URL. This is a precursor to allow custom `prefixes` inside `+native`